src/components/coin-row/FastRequestCoinRow.tsx

```diff
@@ -0,0 +1,143 @@
+import { addHours, differenceInMinutes, isPast } from 'date-fns';
+import lang from 'i18n-js';
+import React, { useCallback, useEffect, useRef, useState } from 'react';
+import { StyleSheet, View } from 'react-native';
+import { useDispatch } from 'react-redux';
+import { ButtonPressAnimation } from '../animations';
+import { FastRequestCoinIcon } from '../coin-icon';
+import { Text } from '@rainbow-me/design-system';
+import { useNavigation } from '@rainbow-me/navigation';
+import { removeRequest } from '@rainbow-me/redux/requests';
+import Routes from '@rainbow-me/routes';
+import { ThemeContextProps } from '@rainbow-me/theme';
+
+const cx = StyleSheet.create({
+  bottomRow: {
+    flexDirection: 'row',
+    justifyContent: 'space-between',
+  },
+  column: {
+    flex: 1,
+    justifyContent: 'space-between',
+    marginLeft: 11,
+  },
+  icon: {
+    justifyContent: 'center',
+  },
+  topRow: {
+    flexDirection: 'row',
+  },
+  wholeRow: {
+    flexDirection: 'row',
+    height: 40,
+    justifyContent: 'space-between',
+    marginVertical: 17,
+    paddingHorizontal: 19,
+  },
+});
+
+const getPercentageOfTimeElapsed = (startDate: Date, endDate: Date) => {
+  const originalDifference = differenceInMinutes(endDate, startDate);
+  const currentDifference = differenceInMinutes(endDate, Date.now());
+
+  return Math.floor((currentDifference * 100) / originalDifference);
+};
+
+export default React.memo(function RequestCoinRow({
+  item,
+  theme,
+}: {
+  item: any;
+  theme: ThemeContextProps;
+}) {
+  const buttonRef = useRef();
+  const dispatch = useDispatch();
+  const { navigate } = useNavigation();
+
+  const [state, setState] = useState<{
+    expirationColor: string;
+    expiresAt: Date | null;
+    percentElapsed: number;
```

Comment:
I would lean towards moving this to a named interface somewhere above the component, hat do you think?

Reply:
Absolutely!